modules/persistent-storage-csi-manila-install-driver.adoc

```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent-storage-csi-manila.adoc
+
+[id="persistent-storage-csi-manila-storage-class_{context}"]
+= Installing the OpenStack Manila CSI Driver Operator
+
+The CSI Manila Operator is not installed in {product-title} by default. Use the following procedure to install and configure this Operator to enable the CSI Manila driver in your cluster.
+
+.Prerequisites
+* You have access to the {product-title} web console.
+* NFS share types are used for provisioning.
+//This prereq won't be required unless there are custom install instructions that are added to OCP docs:
+//* The OpenStack Manila service has been correctly set up and is accessible from the {product-title} cluster.
+
+.Procedure
+
+To install the OpenStack Manila CSI Driver Operator from the web console, follow these steps:
+
+. Log in to the {product-title} web console.
+
+. Navigate to *Operators* -> *OperatorHub*.
+
+. Type *Manila CSI* into the filter box to locate the OpenStack Manila CSI Driver Operator.
+
+. Click *Install*.
+
+. On the *Install Operator* page, be sure that *A specific namespace on the cluster* is selected. Select *openshift-csi-driver-manila-operator* from the drop-down menu.
+
+. Adjust the values for *Update Channel* and *Approval Strategy* to the values that you want.
+
+All of the driver's resources are created in the `openshift-manila-csi-driver` namespace.
+
+Once finished, the OpenStack Manila CSI Driver Operator is listed in the *Installed Operators* section of the web console.
```

Comment:
After that, we need to create ManilaDriver instance to create manila driver, this will create all related components including storage class "csi-manila-ceph", which we can use it for dynamic provision.

Reply:
I was under the impression that this Operator creates the Manila driver and thus a default storage class, similar to AWS EBS Operator: https://github.com/openshift/openshift-docs/pull/22350/files#diff-8e09bc6a796983013e07888f275920e5R15
If that's not the case, can you describe the steps to create the manila driver, please?

Reply:
@bobfuru Yes, it should be a similar step with AWS EBS CSI Operator, but looks it also missed the step to create CSI driver and related resource. If I understand well, actually it is also like the local volume in https://docs.openshift.com/container-platform/4.4/storage/persistent_storage/persistent-storage-local.html#local-volume-cr_persistent-storage-local

Following is install ManilaDriver from UI/CLI, and check the installation for your rreference.
Install ManilaDriver from the UI
1. On the "Installed Operators" page, select and entry "Manila CSI Driver Operator", in the "ManilaDriver" tab, you can see installed ManilaDrivers here.
2. Click "Create ManilaDriver" button, then click "Create" button to create manila driver. please note in "YAML View", the name "cluster" cound not be renamed, and it doesn't support specify a certain namespace.

Install ManilaDriver from the Click
1. Create the ManilaDriver  
Example: ManilaDriver
$ more maniladriver.yaml
apiVersion: csi.openshift.io/v1alpha1
kind: ManilaDriver
metadata:
  name: cluster
  
 please note the name "cluster" cound not be renamed, also it doesn't support specify a certain namespace. 
 
2. Create the ManilaDriver in your OpenShift Container Platform cluster, specifying the file you just created:
$ oc create -f maniladriver.yaml


Verify ManilaDriver installed successfully
1. Verify corresponding resource are created successfully:
$ oc get all -n openshift-manila-csi-driver
NAME                                                         READY   STATUS    RESTARTS   AGE
pod/csi-nodeplugin-nfsplugin-lzvpm                           1/1     Running   0          18h
pod/csi-nodeplugin-nfsplugin-slvg2                           1/1     Running   0          18h
pod/csi-nodeplugin-nfsplugin-xmps9                           1/1     Running   0          18h
pod/openstack-manila-csi-controllerplugin-7d4f5d985b-mw4x5   3/3     Running   0          17h
pod/openstack-manila-csi-nodeplugin-6xchs                    2/2     Running   0          18h
pod/openstack-manila-csi-nodeplugin-bkcmz                    2/2     Running   0          18h
pod/openstack-manila-csi-nodeplugin-rlpps                    2/2     Running   0          18h

NAME                                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/csi-nodeplugin-nfsplugin          3         3         3       3            3           <none>          18h
daemonset.apps/openstack-manila-csi-nodeplugin   3         3         3       3            3           <none>          18h

NAME                                                    READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/openstack-manila-csi-controllerplugin   1/1     1            1           18h

NAME                                                               DESIRED   CURRENT   READY   AGE
replicaset.apps/openstack-manila-csi-controllerplugin-7d4f5d985b   1         1         1       17h

2. Verify storage class created successfully:
$ oc get storageclass csi-manila-ceph
NAME              PROVISIONER                RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
csi-manila-ceph   manila.csi.openstack.org   Delete          Immediate           false                  18h

Reply:
Thank you, @duanwei33 - I have some formatting clean-up still to do, but hoping you can take a look at my updates based on your awesome help with this. 😃

Reply:
@bobfuru The persistent-storage-csi-manila-install-driver looks good to me.
Feel free to update my draft  :)

Reply:
@bobfuru As we decide to use "Manila CSI Driver Operator" as the official name, so I suggest using, I see somewhere need to be updated, WDYT? 
- "Manila CSI Driver Operator" instead of "CSI Driver Manila Operator"
- "Manila CSI Driver " instead of "CSI Manila Driver"
- "manila csi driver" instead of "csi manila driver" or "manila driver" when talking about the csi driver

Reply:
Thanks, @duanwei33 - It's been tricky getting the naming right but I think I have updated everything as you indicated. But the name shows up differently in console/Operator Hub: `CSI Driver Manila Operator`. In some places, when talking about the resource, I went with `ManilaDriver` rather than `Manila CSI Driver`.